variorum@0.4 build fails: SetupJansson.cmake: need explicit JANSSON_DIR

`variorum@0.4` build fails using:
* `spack@develop` (98cf0f5e0719353b3326b73581d5a74c8363009e from `Thu May 13 08:48:04 2021 -0600`)
* Ubuntu 20.04 - GCC 9.3.0
* Ubuntu 18.04 - GCC 7.5.0
* RHEL 8 - GCC 8.3.1
* RHEL 7 - GCC 9.3.0
* x86_64 and ppc64le

Using container: `ecpe4s/ubuntu20.04-runner-x86_64:2021-05-10`
Concrete spec: [variorum-imktxr.spec.yaml.txt](https://github.com/spack/spack/files/6473366/variorum-imktxr.spec.yaml.txt)
Full build log: [variorum-build-out.txt](https://github.com/spack/spack/files/6473374/variorum-build-out.txt)


```
$> spack install -f ./variorum-imktxr.spec.yaml
...
==> Installing variorum-0.4.1-2embhm25xv2owudlkfohxk5ty5igjm2q
==> Fetching https://github.com/llnl/variorum/archive/v0.4.1.tar.gz
######################################################################################################################################################### 100.0%          -=O=- #        #        #          #
==> No patches needed for variorum
==> variorum: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/variorum-0.4.1-2embhm25xv2owudlkfohxk5ty5igjm2q' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/variorum-0.4.1-2embhm25xv2owudlkfohxk5ty5igjm2q/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/variorum-0.4.1-2embhm25xv2owudlkfohxk5ty5igjm2q/lib64;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/hwloc-1.11.9-4wyiypie6vtaf65z37zkryp4pifw7qeq/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libpciaccess-0.16-bob4o5m3uku6vtdil5imasprgy775zg7/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libxml2-2.9.10-u3rkiqzxjxcgkrexrc4mkc6ee6ojkdl5/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/libiconv-1.16-jearpk4xci4zc7dkrza4fufaqfkq7rfl/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/xz-5.2.5-d2ghzztrrzcwshcjijohyn6i5pr5gpp7/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/zlib-1.2.11-smoyzzo2qhzpn6mg6rd3l2p7b23enshg/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/ncurses-6.2-7wdfzbnxtikwxzlcbddvjfoj6mop7ic2/lib;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/numactl-2.0.14-ahg7xfg6caqti66xrblbcmpfcaekxys5/lib' '-DCMAKE_PREFIX_PATH:STRING=/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/cmake-3.20.2-4nleebjsxxpub2ykqrz2dniiyhpwwt5b;/opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/hwloc-1.11.9-4wyiypie6vtaf65z37zkryp4pifw7qeq' '-DBUILD_SHARED_LIBS=ON' '-DBUILD_DOCS=OFF' '-DVARIORUM_DEBUG=OFF' '-DVARIORUM_LOG=OFF' '-DBUILD_TESTS=OFF' '/tmp/root/spack-stage/spack-stage-variorum-0.4.1-2embhm25xv2owudlkfohxk5ty5igjm2q/spack-src/src'

1 error found in build log:
     33    -- Building variorum as a shared library (BUILD_SHARED_LIBS == ON)
     34    -- Fortran Compiler: /opt/spack/lib/spack/env/gcc/gfortran
     35    -- Updated CMAKE_CXX_FLAGS to " -std=c++11"
     36    -- FOUND hwloc using find_library()
     37    --  [*] HWLOC_INCLUDE_DIRS = /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/hwloc-1.11.9-4wyiypie6vtaf65z37zkryp4pifw7qeq/include
     38    --  [*] HWLOC_LIBRARY = /opt/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.5.0/hwloc-1.11.9-4wyiypie6vtaf65z37zkryp4pifw7qeq/lib/libhwloc.so
  >> 39    CMake Error at CMake/thirdparty/SetupJansson.cmake:48 (message):
     40      Jansson support needs explict JANSSON_DIR
     41    Call Stack (most recent call first):
     42      CMake/Setup3rdParty.cmake:7 (include)
     43      CMakeLists.txt:49 (include)
     44
     45

See build log for details:
  /tmp/root/spack-stage/spack-stage-variorum-0.4.1-2embhm25xv2owudlkfohxk5ty5igjm2q/spack-build-out.txt
```

@gonsie @fbudin69500

--- a/var/spack/repos/builtin/packages/variorum/package.py
+++ b/var/spack/repos/builtin/packages/variorum/package.py
@@ -39,6 +39,7 @@ class Variorum(CMakePackage):
     ########################
     depends_on("cmake@2.8:", type="build")
     depends_on("hwloc")
+    depends_on("jansson", type="link")
 
     #########################
     # Documentation related #
@@ -50,6 +51,8 @@ class Variorum(CMakePackage):
     def cmake_args(self):
         spec = self.spec
         cmake_args = []
+
+        cmake_args.append('-DJANSSON_DIR={0}'.format(spec['jansson'].prefix))
 
         if "+shared" in spec:
             cmake_args.append("-DBUILD_SHARED_LIBS=ON")

